STIPS/STIP-009.md

```diff
@@ -0,0 +1,1094 @@
+
+----
+
+> addExtension
+
+ONLY OWNER: Add a new extension that the DelegatedManager can call
+
+```solidity
+function addExtensions(address[] memory _extensions)
+```
+
++ for each extension in _extensions
+  + require that extension state in *extensionAllowList* is *NONE* (has not already been added)
+  + set *extensionAllowlist[extension]* to PENDING
+  + emit *ExtensionAdded* event
+
+----
+
+> removeExtensions
+
+ONLY OWNER: Remove an existing extension tracked by the DelegatedManager.
+
+```solidity
+function removeExtensions(address[] memory _extensions)
+```
+
++ for each extension in _extensions
+  + require that extension state in *extensionAllowList* is *INITIALZED*
+  + delete extension from *extensions* array
+  + set extension state to *NONE* in *extensionAllowList*
+  + call extensions's own *removeExtension* method for manager's setToken
+  + emit *ExtensionRemoved* event
+
+>
+
+----
+
+> addOperators
+
+ONLY OWNER: Add new operator(s) address
+
+```solidity
+function addOperators(address[] memory _operators)
+```
++ for each operator in _operators
+  + require that operator is not already registered in the *operatorAllowList* mapping
+  + add operator to the *operators* array
+  + set *operatorAllowlist[operator]* to `true`
+  + emit *OperatorAdded* event
+
+----
+
+> removeOperators
+
+ONLY OWNER: Remove operator(s) from the allowlist
+
+```solidity
+function removeOperators(address[] memory _operators)
+```
+
++ for each operator in _operators
+  + require that operator is in the *operatorAllowList* mapping
+  + delete operator from the *operators* array
+  + set *operatorAllowlist[operator]* to `false`
+  + emit *OperatorRemoved* event
+
+----
+
+> addAllowedAssets
+
+ONLY OWNER: Add new asset(s) that can be traded to, wrapped to, or claimed
+
+```solidity
+function addAllowedAssets(address[] memory _assets)
+```
++ for each asset in _assets
+  + require that asset is not already registered in the *assetAllowList* mapping
+  + add asset to the *assetAllowList* array
+  + set *assetAllowlist[asset]* to `true`
+  + emit *AllowAssetAdded* event
+
+---
+
+> removeAllowedAssets
+
+ONLY OWNER: Remove asset(s) so that it/they can't be traded to, wrapped to, or claimed
+
+```solidity
+function removeAllowedAssets(address[] memory _assets)
+```
+
++ for each asset in _assets
+  + require that asset is present in the *assetAllowList* mappiing
+  + delete asset from the *allowedAssets* array
+  + set *assetAllowlist[asset]* to `false`
+  + emit AllowedAssetRemoved event
+
+----
+
+> setUseAssetAllowList
+
+ONLY OWNER: Toggles whether or not operator can trade any asset. When false, assetAllowList
+restrictions are ignored.
+
+```solidity
+function setUseAssetAllowList(bool _useAssetAllowList)
+```
+
++ set *useAssetAllowList* to _useAssetAllowList
++ emit *UseAllowListUpdated* event
+
+----
+
+> setOwnerFeeSplit
+
+ONLY OWNER: Sets the *ownerFeeSplit*
+
+```solidity
+function setOwnerFeeSplit(uint256 _ownerFeeSplit)
```

Comment:
Small nit: Let's call this `updateOwnerFeeSplit`, set has a bit more loaded of a name for us.